File: backend/src/main/java/com/ifpb/agendapet/pet/dto/PetResponseDTO.java
package com.ifpb.agendapet.pet.dto;

import com.ifpb.agendapet.appointment.dto.AppointmentResponseDTO;
import com.ifpb.agendapet.pet.PetSpecies;
import com.ifpb.agendapet.shared.enums.GenderEnum;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record PetResponseDTO(UUID id, String name, Double weight, UUID tutor_id, String tutor_name, GenderEnum gender, LocalDate birthday, PetSpecies species, String breed, String description, List<AppointmentResponseDTO> appointments, LocalDateTime created_at, LocalDateTime updated_at) {
}

File: frontend/src/app/models/DTO/pet-response-DTO.ts
export interface PetResponseDTO {
  id: string;
  name: string;
  avatarUrl?: string;
  weight: number;
  tutor_id: string;
  tutor_name: string;
  gender: PetGenderEnum;
  birthday: string;
  species: PetSpecies;
  breed: string;
  description: string;
  appointments?: AppointmentBasicDTO[];
  created_at: string;
  updated_at: string;
}

export enum PetGenderEnum {
  MALE = 'MALE',
  FEMALE = 'FEMALE',
}

export enum PetSpecies {
  CAT = 'CAT',
  DOG = 'DOG',
  BIRD = 'BIRD',
  RODENT = 'RODENT',
  REPTILE = 'REPTILE',
  OTHER = 'OTHER',
}

export interface AppointmentBasicDTO {
  id: string;
  scheduled_at: string;
  type: string;
  status: string;
}
